src/index.js

```diff
@@ -27,25 +27,20 @@ var defaultOptions = {
  */
 SavitzkyGolay.prototype.calc = function (options) {
     options = options || {};
-    this.options = {};
-    for (var o in defaultOptions) {
-        if (options.hasOwnProperty(o)) {
-            this.options[o] = options[o];
-        } else {
-            this.options[o] = defaultOptions[o];
-        }
-    }
-    if ((this.options.windowSize % 2 === 0) || (this.options.windowSize < 5))
+    for (var o in defaultOptions)
+        if (!(options.hasOwnProperty(o)))
+            options[o] = defaultOptions[o];
+    if ((options.windowSize % 2 === 0) || (options.windowSize < 5))
         throw new RangeError('Invalid window size');
-    if ((this.options.derivative < 0) || (this.options.polynomial < 1))
+    if ((options.derivative < 0) || (options.polynomial < 1))
         throw new RangeError('Number too small');
```

Comment:
I would check them on a different line so that the error message can be like "derivative option must be a positive number"